app/src/main/java/com/cornellappdev/coursegrab/CourseDetailsActivity.kt

```diff
@@ -0,0 +1,160 @@
+package com.cornellappdev.coursegrab
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.cornellappdev.coursegrab.models.ApiResponse
+import com.cornellappdev.coursegrab.models.Course
+import com.cornellappdev.coursegrab.models.SearchResult
+import com.cornellappdev.coursegrab.models.Section
+import com.cornellappdev.coursegrab.networking.Endpoint
+import com.cornellappdev.coursegrab.networking.Request
+import com.cornellappdev.coursegrab.networking.addTracking
+import com.cornellappdev.coursegrab.networking.removeTracking
+import com.google.gson.reflect.TypeToken
+import kotlinx.android.synthetic.main.activity_course_details.*
+import kotlinx.android.synthetic.main.activity_search.back_btn
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class CourseDetailsActivity : AppCompatActivity() {
+    private lateinit var sectionsRecyclerView: RecyclerView
+    private lateinit var sectionsViewAdapter: RecyclerView.Adapter<*>
+    private lateinit var sectionsViewManager: RecyclerView.LayoutManager
+
+    private val preferencesHelper: PreferencesHelper by lazy {
+        PreferencesHelper(this)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_course_details)
+
+        var courseDetails: SearchResult = intent.getParcelableExtra("courseDetails") as SearchResult
+
+        header_textView.text = "${courseDetails.subject_code} ${courseDetails.course_num}"
+
+        course_title.text = courseDetails.title
+        course_details.text =
+            if (courseDetails.sections.isNotEmpty()) courseDetails.sections.first().instructors.first() else "To Be Assigned"
+
+        // Available Courses Adapter
+        sectionsViewManager = LinearLayoutManager(this@CourseDetailsActivity)
+        sectionsViewAdapter = SectionAdapter(courseDetails.sections, this@CourseDetailsActivity)
+
+        sectionsRecyclerView = findViewById<RecyclerView>(R.id.sections_recyclerview).apply {
```

Comment:
nit: don't need `findViewById` in Kotlin

Reply:
Didn't even notice that. Well spotted, I'll change that across all the activities with recyclers.